streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -216,12 +216,14 @@ synchronized void unlock(final TaskId taskId) throws IOException {
         final LockAndOwner lockAndOwner = locks.get(taskId);
         if (lockAndOwner != null && lockAndOwner.owningThread.equals(Thread.currentThread().getName())) {
             locks.remove(taskId);
-            lockAndOwner.lock.release();
-            log.debug("{} Released state dir lock for task {}", logPrefix(), taskId);
-
-            final FileChannel fileChannel = channels.remove(taskId);
-            if (fileChannel != null) {
-                fileChannel.close();
+            try {
```

Comment:
Hello @asutosh936, I do not know if you already did, maybe we can consider using the `try-with-resources` statement here, like:

```
try (FileChannel fileChannel = channels.remove(taskId)) {
    lockAndOwner.lock.release();
    log.debug("{} Released state dir lock for task {}", logPrefix(), taskId);
}
```

What do you think?

Reply:
Yes... `we can use try-with-resources`. Implemented review comment.